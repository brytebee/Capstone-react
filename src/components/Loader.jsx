import React from 'react';
import ScaleLoader from 'react-spinners/ScaleLoader';

const Loader = () => (
  <div className="spinner">
    <ScaleLoader color="#4369B2" />
  </div>
);

export default Loader;
