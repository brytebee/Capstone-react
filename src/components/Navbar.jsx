import React from 'react';
import { useNavigate } from 'react-router-dom';
import { IoIosArrowBack } from 'react-icons/io';
import { HiOutlineMicrophone } from 'react-icons/hi';
import { IoSettingsOutline } from 'react-icons/io5';

const Navbar = () => {
  const navigate = useNavigate();

  const handleBack = () => {
    navigate('/');
  };
  return (
    <div className="navbarista flex-btw">
      <div className="flex-btw">
        <IoIosArrowBack onClick={handleBack} className="back-arrow" />
      </div>
      <div>
        <HiOutlineMicrophone />
        <IoSettingsOutline />
      </div>
    </div>
  );
};

export default Navbar;
