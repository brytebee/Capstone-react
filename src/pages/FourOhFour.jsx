import React from 'react';

const img = 'https://i.pinimg.com/originals/0e/c0/db/0ec0dbf1e9a008acb9955d3246970e15.gif';

const FourOhFour = () => (
  <div>
    <h2>Oh Oh!</h2>
    <p>The page you are trying to access is unavailable at the moment...</p>
    <img className="four-oh-four" src={img} alt="Error dispaly" />
  </div>
);

export default FourOhFour;
