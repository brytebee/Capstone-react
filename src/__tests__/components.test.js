import { cleanup } from '@testing-library/react';
import renderer from 'react-test-renderer';
import Loader from '../components/Loader';
import Navbar from '../components/Navbar';
import FourOhFour from '../pages/FourOhFour';

const mockDispatch = jest.fn();
jest.mock('react-redux', () => ({
  useSelector: jest.fn(),
  useDispatch: () => mockDispatch,
}));

const mockedUsedNavigate = jest.fn();
jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useNavigate: () => mockedUsedNavigate,
}));

afterEach(cleanup);

describe('Snapshot of Covid19STAT component', () => {
  it('Snapshot of the Loader Component', () => {
    const tree = renderer.create(<Loader />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Snapshot of the Navbar Component', () => {
    const tree = renderer.create(<Navbar />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Snapshot of the FourOhFour Component', () => {
    const tree = renderer.create(<FourOhFour />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
