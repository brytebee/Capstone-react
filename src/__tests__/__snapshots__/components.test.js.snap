// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot of Covid19STAT component Snapshot of the FourOhFour Component 1`] = `
<div
  className="four-oh-four"
>
  <h2>
    Oh Oh!
  </h2>
  <p>
    The page you are trying to access is unavailable at the moment...
  </p>
  <img
    alt="Error dispaly"
    className="four-oh-four-img"
    src="https://i.pinimg.com/originals/0e/c0/db/0ec0dbf1e9a008acb9955d3246970e15.gif"
  />
</div>
`;

exports[`Snapshot of Covid19STAT component Snapshot of the Loader Component 1`] = `
<div
  className="spinner"
>
  <span
    className="css-1xdhyk6"
  >
    <span
      className="css-1ionjmu"
    />
    <span
      className="css-1dx4to7"
    />
    <span
      className="css-8elkgk"
    />
    <span
      className="css-1ip0fzg"
    />
    <span
      className="css-of60g4"
    />
  </span>
</div>
`;

exports[`Snapshot of Covid19STAT component Snapshot of the Navbar Component 1`] = `
<div
  className="navbarista flex-btw"
>
  <div
    className="flex-btw"
  >
    <svg
      fill="currentColor"
      height="1em"
      onClick={[Function]}
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 512 512"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M217.9 256L345 129c9.4-9.4 9.4-24.6 0-33.9-9.4-9.4-24.6-9.3-34 0L167 239c-9.1 9.1-9.3 23.7-.7 33.1L310.9 417c4.7 4.7 10.9 7 17 7s12.3-2.3 17-7c9.4-9.4 9.4-24.6 0-33.9L217.9 256z"
      />
    </svg>
  </div>
  <div>
    <svg
      fill="none"
      height="1em"
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 24 24"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M19 11a7 7 0 01-7 7m0 0a7 7 0 01-7-7m7 7v4m0 0H8m4 0h4m-4-8a3 3 0 01-3-3V5a3 3 0 116 0v6a3 3 0 01-3 3z"
        strokeLinecap="round"
        strokeLinejoin="round"
        strokeWidth="2"
      />
    </svg>
    <svg
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 512 512"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M262.29 192.31a64 64 0 1057.4 57.4 64.13 64.13 0 00-57.4-57.4zM416.39 256a154.34 154.34 0 01-1.53 20.79l45.21 35.46a10.81 10.81 0 012.45 13.75l-42.77 74a10.81 10.81 0 01-13.14 4.59l-44.9-18.08a16.11 16.11 0 00-15.17 1.75A164.48 164.48 0 01325 400.8a15.94 15.94 0 00-8.82 12.14l-6.73 47.89a11.08 11.08 0 01-10.68 9.17h-85.54a11.11 11.11 0 01-10.69-8.87l-6.72-47.82a16.07 16.07 0 00-9-12.22 155.3 155.3 0 01-21.46-12.57 16 16 0 00-15.11-1.71l-44.89 18.07a10.81 10.81 0 01-13.14-4.58l-42.77-74a10.8 10.8 0 012.45-13.75l38.21-30a16.05 16.05 0 006-14.08c-.36-4.17-.58-8.33-.58-12.5s.21-8.27.58-12.35a16 16 0 00-6.07-13.94l-38.19-30A10.81 10.81 0 0149.48 186l42.77-74a10.81 10.81 0 0113.14-4.59l44.9 18.08a16.11 16.11 0 0015.17-1.75A164.48 164.48 0 01187 111.2a15.94 15.94 0 008.82-12.14l6.73-47.89A11.08 11.08 0 01213.23 42h85.54a11.11 11.11 0 0110.69 8.87l6.72 47.82a16.07 16.07 0 009 12.22 155.3 155.3 0 0121.46 12.57 16 16 0 0015.11 1.71l44.89-18.07a10.81 10.81 0 0113.14 4.58l42.77 74a10.8 10.8 0 01-2.45 13.75l-38.21 30a16.05 16.05 0 00-6.05 14.08c.33 4.14.55 8.3.55 12.47z"
        fill="none"
        strokeLinecap="round"
        strokeLinejoin="round"
        strokeWidth="32"
      />
    </svg>
  </div>
</div>
`;
